src/librustc/session/mod.rs

```diff
@@ -336,6 +336,10 @@ impl NodeIdAssigner for Session {
     fn peek_node_id(&self) -> NodeId {
         self.next_node_id.get().checked_add(1).unwrap()
     }
+
+    fn span_fatal(&self, sp: MultiSpan, msg: &str) -> ! {
```

Comment:
Done (except this code path is now untested because the parse error was made fatal).